File: CS/Scaffolding.ManyToMany/ViewModels/CourseViewModel.Extensions.cs
using Scaffolding.ManyToMany.Common.ViewModel;
using Scaffolding.ManyToMany.Model;
using Scaffolding.ManyToMany.SchoolContextDataModel;
using System;
using System.Linq;


namespace Scaffolding.ManyToMany.ViewModels {
    partial class CourseViewModel {
        protected override void RefreshLookUpCollections(int key) {
            base.RefreshLookUpCollections(key);
            CourseStudents = CreateAddRemoveDetailEntitiesViewModel(x => x.Students, x => x.Students);
        }
        public virtual AddRemoveDetailEntitiesViewModel<Course, int, Student, int, ISchoolContextUnitOfWork> CourseStudents { get; protected set; }
    }
}

File: VB/Scaffolding.ManyToMany/ViewModels/Course/CourseViewModel.Extensions.vb
Imports Scaffolding.ManyToMany.Common.ViewModel
Imports Scaffolding.ManyToMany.Model
Imports Scaffolding.ManyToMany.SchoolContextDataModel
Imports System
Imports System.Linq


Namespace Scaffolding.ManyToMany.ViewModels
    Partial Public Class CourseViewModel
        Protected Overrides Sub RefreshLookUpCollections(ByVal raisePropertyChanged As Boolean)
            MyBase.RefreshLookUpCollections(raisePropertyChanged)
            CourseStudents = CreateAddRemoveDetailEntitiesViewModel(Function(x) x.Students, Function(x) x.Students)
        End Sub
        Private privateCourseStudents As AddRemoveDetailEntitiesViewModel(Of Course, Integer, Student, Integer, ISchoolContextUnitOfWork)
        Public Overridable Property CourseStudents() As AddRemoveDetailEntitiesViewModel(Of Course, Integer, Student, Integer, ISchoolContextUnitOfWork)
            Get
                Return privateCourseStudents
            End Get
            Protected Set(ByVal value As AddRemoveDetailEntitiesViewModel(Of Course, Integer, Student, Integer, ISchoolContextUnitOfWork))
                privateCourseStudents = value
            End Set
        End Property
    End Class
End Namespace